### PR TITLE
Search: improve results for simple queries

### PR DESCRIPTION
**Pull request by stsewd**
SimpleQueryString don't allow us to set an implicit value for fuzziness,
but is still useful for advanced queries.
This allows us to support both.

Results look good in local testing and comparing them with the results
from production.

### CODE REVIEW

File: readthedocs/search/tests/test_xss.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 class TestXSS:
 
     def test_facted_page_xss(self, client, project):
-        query = 'XSS'
+        query = '"XSS"'
```
Comment (ericholscher) [inline readthedocs/search/tests/test_xss.py:11]: Why did this change?

File: readthedocs/search/tests/test_api.py [hunk 2]
```diff
@@ -439,6 +439,43 @@ def test_search_correct_link_for_index_page_subdirectory_htmldir_projects(self,
...
+        # Query with a typo should return results
+        search_params = {
+            'project': project.slug,
+            'version': version.slug,
+            'q': 'indx',
+        }
+        resp = self.get_search(api_client, search_params)
+        assert resp.status_code == 200
+
+        assert len(resp.data['results']) > 0
```
Comment (ericholscher) [inline readthedocs/search/tests/test_api.py:455]: We should probably validate it returns the result we expect (`index in resp.data['results']` or something)

File: readthedocs/search/faceted_search.py [hunk 3]
```diff
@@ -55,6 +55,49 @@ def __init__(self, query=None, filters=None, user=None, **kwargs):
...
+           the query contains any of them should be enough to determinate if
+           it's an advanced query.
+
+        Simple query syntax:
+
+        https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-simple-query-string-query.html#simple-query-string-syntax
+        """
+        tokens = {'+', '|', '-', '"' , '*', '(', ')', '~'}
+        query_tokens = set(query)
+        return not tokens.isdisjoint(query_tokens)
```
Comment (ericholscher) [inline readthedocs/search/faceted_search.py:99]: Fancy.

File: readthedocs/search/faceted_search.py [hunk 4]
```diff
@@ -71,10 +114,9 @@ def query(self, search, query):
 
         # need to search for both 'and' and 'or' operations
         # the score of and should be higher as it satisfies both or and and
-
         for operator in self.operators:
-            query_string = SimpleQueryString(
-                query=query, fields=self.fields, default_operator=operator
+            query_string = self._get_text_query(
+                query, self.fields, operator,
```
Comment (ericholscher) [inline readthedocs/search/faceted_search.py:119]: I like passing kwargs to all functions, it makes things more explicit in my experience.

**Comment by ericholscher** [comment]
I wonder if it's worth putting this behind a feature flag for testing -- I do think it might cause some confusion, especially on the .com

File: readthedocs/search/tests/test_xss.py (hunk 1 above)
Comment (stsewd) [reply readthedocs/search/tests/test_xss.py:11]: It was matching more results

https://github.com/readthedocs/readthedocs.org/blob/7fd916547c0ac8dd70eaddd4a07a2be0846ab062/readthedocs/search/tests/test_xss.py#L19-L19

We do check the content from the result below, so it's safe
https://github.com/readthedocs/readthedocs.org/blob/fb5b8c8a8a29bdaf9123806f6327ccc101419c3c/readthedocs/search/tests/test_xss.py#L32-L32

**Comment by stsewd** [comment]
>  I do think it might cause some confusion, especially on the .com

Not sure what would be confusing. Actually I'm getting results more quickly in the search as you type, since I don't need to type the full word or remember the exact word in english. But I'm fine putting this behind a feature flag, let me know if you want to go that way.

**Comment by ericholscher** [comment]
I think it's more that the results will change, so if people are used to a certain result, it might not show up as high in the results.

Do you find it lists all exact matches prior to partial? It seems like the XSS test suggests that isn't the case?

**Comment by stsewd** [comment]
> I think it's more that the results will change, so if people are used to a certain result, it might not show up as high in the results.

Ok, got it. A feature flag makes sense them, we could see how the results changed in our docs (but in local testing with my frequent queries was all good) 

> Do you find it lists all exact matches prior to partial? It seems like the XSS test suggests that isn't the case?

The XSS query was listing more results without `""`, not fewer. Is that is what you mean.

**Comment by ericholscher** [comment]
> The XSS query was listing more results without "", not fewer. If that is what you mean.

Right, but presumably the additional results were partial matches? My question is: *do all full matches return higher in results than any partial matches* -- because that would be an issue.

**Comment by stsewd** [comment]
Ok, I just checked, the first results is what we expect


```
(Pdb) pp hits[0]
{'_id': '5',
 '_index': 'test_page_index',
 '_score': 4.838786,
 '_source': {'build': None,
             'commit': '5',
             'domains': [{'role_name': 'py:function', 'anchor': 'celery.utils.depreca...},
                         {'role_name': 'py:function', 'anchor': 'celery.utils.depreca...}],
             'full_path': 'support.html',
             'path': 'support',
             'project': 'docs',
             'sections': [{'id': 'usage-questions', 'title': 'Usage Questions', 'conte...},
                          {'id': 'community-support', 'title': 'Community Support', 'c...},
                          {'id': 'commercial-support', 'title': 'Commercial Support', ...}],
             'title': 'Support',
             'version': 'latest'},
 '_type': 'doc',
 'inner_hits': {'domains': <Response: {}>,
                'sections': <Response: [{'_index': 'test_page_index', '_type': 'doc', '_id': '5', '_...}]>}}

```

That's from the page we expect https://github.com/readthedocs/readthedocs.org/blob/master/readthedocs/search/tests/data/docs/support.json#L18


I have added the feature flag on the indoc-search, we can't add the feature flag to the dashboard search bc isn't related to a project but to a user (it defaults to the old behavior, so we are fine)

**Pull request merged**